modules/config-preconfigure-automation.adoc

```diff
@@ -31,6 +31,53 @@ SUPER_USERS:
 ...
 ----
 
+=== Resetting superuser passwords on {productname} Operator deployments
+
+Use the following procedure to reset a superuser's password. 
+
+.Prerequisites 
+
+* You have configured a superuser. 
+
+.Procedure 
+
+* Enter the following command to generate a new password hash. This example generates a hash for the plaintext password `NewPassword`. 
++
+[source,terminal]
+----
+$ python3 -c 'import bcrypt; print(bcrypt.hashpw(b"NewPassword", bcrypt.gensalt(12)).decode("utf-8"))'
+----
++
+Example output:
+[source,terminal]
+----
+f2O4AWmVaq8-SMd9Rs2-JwsbCEsuyOEOx-DBkC1uq2UfWvsgpQMqDtjwXOTzYDpFLf60xtSlHNBaRU-6
+----
+
+. Enter the following commands to change the `password_hash` for a specific superuser:
++
+[source,terminal]
+----
+$ oc rsh quayadmin-reg-quay-database-7b8cc8856d-44qhq
+----
++
+[source,terminal]
+----
+sh-4.4$ echo "UPDATE public.user SET password_hash = 'f2O4AWmVaq8-SMd9Rs2-JwsbCEsuyOEOx-DBkC1uq2UfWvsgpQMqDtjwXOTzYDpFLf60xtSlHNBaRU-6' where username = '$quayadmin'" |  psql -U sayali-reg-quay-database -h 10.129.4.8 quayadmin-reg-quay-database
+----
+
+. When prompted, enter the new password for your superuser, for example:
+[source,terminal]
+----
+Password for user quayadmin-reg-quay-database: <new_password>
+----
++
+Example output:
+[source,terminal]
+----
+UPDATE 1
+----
```

Comment:
After finished above steps, my login crashed .   I found below problems. 

* run `python3 -c 'import bcrypt; print(bcrypt.hashpw(b"NewPassword", bcrypt.gensalt(12)).decode("utf-8"))'` more times, the outputs are different.  Does it work as expected?
```
$ python3 -c 'import bcrypt; print(bcrypt.hashpw(b"NewPassword", bcrypt.gensalt(12)).decode("utf-8"))'
$2b$12$ADGJg0dOA55xHnMvPpEwi.PxVXCcSY7IxBNaYlm/aKH7pQwmcBfEO

$ python3 -c 'import bcrypt; print(bcrypt.hashpw(b"NewPassword", bcrypt.gensalt(12)).decode("utf-8"))'
$2b$12$vV2eeEsY/ZlQ0TErvitT/.lgZ8B./JIMzrwYF7/.iLYyD8N5Tkl.y
```

* After updated database. the `password_hash` in table `public.user` is not the string generated by python3
```
$ oc get pod -l quay-component=postgres 
NAME                                          READY   STATUS    RESTARTS   AGE
quayregistry-quay-database-5876bd5d74-5h8xr   1/1     Running   0          3h27m

$ oc -n quay-enterprise rsh $(oc get pod -l app=quay -o NAME -n quay-enterprise|head -n 1) cat /conf/stack/config.yaml|grep DB_URI
DB_URI: postgresql://quayregistry-quay-database:jNLy1z8mk3u9uQ8HtvbFWa4f2B-T6K0C4SPh0WT1PvLRjk6iQkmKaiNMxy8e6GwxwYRtgjsN70LLNBQM@quayregistry-quay-database:5432/quayregistry-quay-database

$ oc rsh quayregistry-quay-database-5876bd5d74-5h8xr 
sh-4.4$ psql -U quayregistry-quay-database  -d quayregistry-quay-database
psql (10.21)
Type "help" for help.

quayregistry-quay-database=>   select * from public.user;
 id |                 uuid                 |        username         |                        password_hash                    
     |               email               | verified | stripe_id | organization | robot | invoice_email | invalid_login_attempts
 |     last_invalid_login     | removed_tag_expiration_s | enabled | invoice_email_address | company | family_name | given_name
 | location | maximum_queued_builds_count |       creation_date        | last_accessed 
----+--------------------------------------+-------------------------+---------------------------------------------------------
-----+-----------------------------------+----------+-----------+--------------+-------+---------------+-----------------------
-+----------------------------+--------------------------+---------+-----------------------+---------+-------------+-----------
-+----------+-----------------------------+----------------------------+---------------
  1 | f7651dbc-c95a-419e-9cba-2aaa4a503fe3 | whuquay                 | $2b$12$bA5e7vseF3i8z90wRprp4.fHzvhcI8Nqsq5UlWUfMRLL3.rWh
LqGW | whuquay@bogus.com                 | t        |           | f            | f 

quayregistry-quay-database=> \q

sh-4.4$ echo "UPDATE public.user SET password_hash = '$2b$12$vV2eeEsY/ZlQ0TErvitT/.lgZ8B./JIMzrwYF7/.iLYyD8N5Tkl.y' where username = 'whuquay'" |  psql -U quayregistry-quay-database  -d quayregistry-quay-database
UPDATE 1

sh-4.4$ psql -U quayregistry-quay-database  -d quayregistry-quay-database
psql (10.21)
Type "help" for help.

quayregistry-quay-database=> select * from public.user;
 id |                 uuid                 |        username         |                        password_hash                    
     |               email               | verified | stripe_id | organization | robot | invoice_email | invalid_login_attempts
 |     last_invalid_login     | removed_tag_expiration_s | enabled | invoice_email_address | company | family_name | given_name
 | location | maximum_queued_builds_count |       creation_date        | last_accessed 
----+--------------------------------------+-------------------------+---------------------------------------------------------
-----+-----------------------------------+----------+-----------+--------------+-------+---------------+-----------------------
-+----------------------------+--------------------------+---------+-----------------------+---------+-------------+-----------
-+----------+-----------------------------+----------------------------+---------------
  1 | f7651dbc-c95a-419e-9cba-2aaa4a503fe3 | whuquay                 | b2/ZlQ0TErvitT/.lgZ8B./JIMzrwYF7/.iLYyD8N5Tkl.y         
     | whuquay@bogus.com                 | t        |           | f            | f     | f             |                      1
 | 2022-11-17 16:34:09.384421 |                  1209600 | t       |                       |         |             |           
 |          |                             | 2022-11-17 13:49:31.599391 | 
```
I thought the password_hash was not changed correctly.

Reply:
It is normal bcrypt behavior that :  bcrypt hash is different every time
https://stackoverflow.com/questions/52121168/bcrypt-encryption-different-every-time-with-same-input